presto-main/src/test/java/com/facebook/presto/operator/aggregation/fixedhistogram/TestFixedDoubleBreakdownHistogram.java

```diff
@@ -0,0 +1,173 @@
+                new FixedDoubleBreakdownHistogram(200, 3.0, 4.0);
+
+        histogram.add(3.1, 100.0);
+        histogram.add(3.8, 200.0);
+        histogram.add(3.1, 100.0);
+        assertEquals(
+                Streams.stream(histogram.iterator()).mapToDouble(c -> c.getWeight()).sum(),
+                300.0);
+    }
+
+    @Test
+    public void testEqualValuesDifferentWeights()
+    {
+        FixedDoubleBreakdownHistogram histogram =
+                new FixedDoubleBreakdownHistogram(200, 3.0, 4.0);
+
+        histogram.add(3.5, 0.2, 1);
+        histogram.add(3.5, 0.4, 1);
+        histogram.add(3.5, 0.3, 2);
+        Streams.stream(histogram.iterator()).forEach(c -> {
+            assertTrue(c.getLeft() <= 3.5);
+            assertTrue(c.getRight() >= 3.5);
+            assertTrue(c.getLeft() < c.getRight());
+        });
+        assertEquals(
+                Streams.stream(histogram.iterator()).mapToLong(c -> 1).sum(),
+                3);
+        assertEquals(
+                Streams.stream(histogram.iterator()).mapToDouble(c -> c.getWeight()).sum(),
+                0.9);
+        assertEquals(
+                Streams.stream(histogram.iterator()).mapToLong(c -> c.getCount()).sum(),
+                4);
+    }
+
+    @Test
+    public void testMassive()
+    {
+        FixedDoubleBreakdownHistogram histogram =
+                new FixedDoubleBreakdownHistogram(100, 0.0, 1.0);
+
+        ArrayList<Double> values = new ArrayList<>();
+        ArrayList<Double> weights = new ArrayList<>();
+
+        IntStream.range(0, 1000000).forEach(i -> {
+            double value = Math.random();
+            double weight = ((int) (10 * Math.random())) / 10.0;
+            values.add(value);
+            weights.add(weight);
+            histogram.add(value, weight, 1);
+        });
+
+        Streams.stream(histogram.iterator()).forEach(b -> {
+            long count = IntStream.range(0, values.size())
+                    .filter(i -> b.getLeft() < values.get(i) && values.get(i) <= b.getRight() && b.getWeight() == weights.get(i))
+                    .count();
+            assertEquals(b.getCount(), count);
+        });
+    }
+
+    @Test
+    public void testMassiveMerge()
+    {
+        ArrayList<Double> values = new ArrayList<>();
+        ArrayList<Double> weights = new ArrayList<>();
+
+        FixedDoubleBreakdownHistogram left =
+                new FixedDoubleBreakdownHistogram(100, 0.0, 1.0);
+        IntStream.range(0, 100).forEach(i -> {
+            double value = Math.random();
+            double weight = Math.random();
+            values.add(value);
+            weights.add(weight);
+            left.add(value, weight, 1);
+        });
+
+        FixedDoubleBreakdownHistogram right =
+                new FixedDoubleBreakdownHistogram(100, 0.0, 1.0);
+        ArrayList<Double> rightValues = new ArrayList<>();
+        ArrayList<Double> rightWeights = new ArrayList<>();
+        IntStream.range(0, 100).forEach(i -> {
```

Comment:
nit: use for-loop.

I found when the body is not concise, using for-loop usually look better than stream :)